file/file.go

```diff
@@ -57,3 +62,23 @@ func (f *singleNodeFile) Read(p []byte) (int, error) {
 	f.offset += n
 	return n, nil
 }
+
+func (f *singleNodeFile) Seek(offset int64, whence int) (int64, error) {
```

Comment:
nit: I'd prefer to keep the methods on a separate type, even if that separate type embeds `singleNodeFile` - then we ensure that the users properly use `AsLargeBytes` and do not rely on the offset state being part of the node itself. Having the methods on the node directly can also encourage using them directly and without `AsLargeBytes`, which can lead to bugs down the line with e.g. `any` or kinded unions.

Arguably that also applies to the existing `Read` node, so perhaps there's a reason you've made this `ipld.Node` implementation stateful. I can't say that's definitely wrong, but I think it would be best to keep the ReadSeeker methods behind the AsT method.